src/controllers/util/conversionStrategies/strategyV1ToV2.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  SourceInput,
+  SourceInputConversionResult,
+  SourceInputV2,
+  SourceRequestV2,
+} from '../../../types';
+import { VersionConversionStrategy } from './abstractions';
+
+export class StrategyV1ToV2 extends VersionConversionStrategy<SourceInput, SourceInputV2> {
+  convert(sourceEvents: SourceInput[]): SourceInputConversionResult<SourceInputV2>[] {
+    // Currently this is not being used
+    // Hold off on testing this until atleast one v2 source has been implemented
+    return sourceEvents.map((sourceEvent) => {
+      try {
+        const sourceRequest: SourceRequestV2 = {
+          method: '',
+          url: '',
+          proto: '',
+          headers: {},
+          query_parameters: {},
+          body: JSON.stringify(sourceEvent.event),
```

Comment:
should we accept types string, array, and object as body to give flexibility and avoid this redundant stringify and parse

Reply:
@koladilip To make the body extendable in the future ( beyond JSON -> xml etc) accepting a string is reasonable in my opinion.
For the time being there is no logic to see which type as rudder-server only accepts json. In the next iterations, rudder-server will accept any input and transformer will parse it using headers to understand the content type. For this flexibility, string is better?

Let me know your thoughts.

Check [this document](https://www.notion.so/rudderstacks/Webhook-Sources-V2-Spec-Upgrade-109f2b415dd0805db37fc8e76556103f?pvs=4#128f2b415dd0809085dccb19951b6ebd) about what the state of body is and should be in a proper context.

Reply:
what I am trying to say is we can handle logic based on type, if the data is a string then try parsing, and if it is a Javascript object then directly use it

Reply:
@koladilip 
Some readability issues with making this type based logic:
- We are making an extra parse only in `StrategyV1ToV2` (which will ideally be used very less)
- This will force us to write new V2 spec sources with checking the type of body and then parsing it along with content-type based parsing which complicates the code and makes in unreadable. (ROI seems less)
- In V2ToV0 and V2ToV1 we will have to strong type it eventhough it will always be string (`const v0Event = JSON.parse(sourceEvent.request.body) as string;`)

If we find that managing types is more valuable here then we can go ahead with it. Let me know if there is anything important I might be missing.